internal/typeinfo/validate.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under Apache 2.0, see LICENCE file for details.
+
+package typeinfo
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func ValidateInputs(args []any) (map[reflect.Type]reflect.Value, error) {
```

Comment:
Comments on these methods please.

I'd consider a type alias for `map[reflect.Type]reflect.Value`

Reply:
Added alias and comments.